### DISCUSSION TITLE
Parse text column by column?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [jsvine/pdfplumber#1039] [Ask for help with specific PDFs]
Hello - is it somehow possible to extract the attached pdf column by column?
(when i read this pdf-file using extract_text() the text is outputted line by line - but i would need it column by column so i get all the person-information in the correct order)

[inp.pdf](https://github.com/jsvine/pdfplumber/files/13256141/inp.pdf)

**Comment by jsvine** [comment]
This is something where you'll want to write some custom logic. I'd look something like this:

1. Identify the x-positions separating the columns — either by hardcoding them, or by examining the characteristics of the objects in `page.lines`.
2. Use those positions to define four bounding boxes, one for each column on the page
3. Use `cropped = page.crop(bbox)` to zoom in on each bounding box, and then `cropped.extract_text()` to get the column's text
